src/compiler/checker.ts

```diff
@@ -22391,10 +22360,8 @@ namespace ts {
                 }
             }
 
-            if (!isRemovedPropertyFromObjectSpread(node.kind === SyntaxKind.Identifier ? node.parent : node)) {
-                const message = isTypeDeclaration(declaration) ? Diagnostics._0_is_declared_but_never_used : Diagnostics._0_is_declared_but_its_value_is_never_read;
-                addDiagnostic(UnusedKind.Local, createDiagnosticForNodeSpan(getSourceFileOfNode(declaration), declaration, node, message, name));
-            }
+            const message = isTypeDeclaration(declaration) ? Diagnostics._0_is_declared_but_never_used : Diagnostics._0_is_declared_but_its_value_is_never_read;

```

Comment:
Separate.

Reply:
and if this is handled with a different change, then do we still need this one?

Reply:
I think so, even if we correctly delete every element from the list, that's different from deleting the entire list.

Reply:
that is another issue.. deleting the entire list may not be correct in the general case.. for instance `var { x } = foo()` if `foo` had side effects, then deleting it would be wrong.. 

One option is to convert it to `var {} = foo()` which is what i was thinking about when i asked about commas earlier. but i do agree that this solution would be ugly. 

Alternatively, we gonna just make it `foo()`, and if the expression was a single identifier or a property access we can remove it. We have in the checker a function `isSideEffectFree`, we could leverage this to decide whether to delete the whole statement or not.

Reply:
Another question, how does this apply to functions. I see in this change you are only handling variable statements..

Reply:
> for instance var { x } = foo() if foo had side effects, then deleting it would be wrong..

But that's exactly what we do for `var x = foo()`... I think it's better to assume that any side effects are unwanted, since the function was being called for a result *before* and not merely for a side effect -- the situation is similar to #17624.

Reply:
#23970